packages/scanner/src/cli/scan/watchScan.ts

```diff
@@ -44,13 +47,54 @@ function isAncestorPath(ancestor: string, descendant: string): boolean {
   return !path.relative(ancestor, descendant).startsWith('..');
 }
 
+export type BatchedTelemetry = {
+  scanResults: ScanResults;
+  msElapsed: number;
+  msInserted: number;
+};
+
 export class Watcher {
   config?: TimestampedConfiguration;
   appmapWatcher?: chokidar.FSWatcher;
   appmapPoller?: chokidar.FSWatcher;
   configWatcher?: chokidar.FSWatcher;
+  batchedTelemetry: BatchedTelemetry[] = [];
+  sendPeriodMilliseconds = 5000;
+  interval: NodeJS.Timer;
+
+  constructor(private options: WatchScanOptions) {
+    this.interval = setInterval(() => {
+      this.sendScanResultsTelemetryInABatch();
+    }, this.sendPeriodMilliseconds);
+    // must unref because there's no condition under which the timer
+    // will stop; else the process doesn't exit
+    this.interval.unref();
+  }
 
-  constructor(private options: WatchScanOptions) {}
+  public sendScanResultsTelemetryInABatch() {
+    const dateNow = Date.now();
+    if (
+      this.batchedTelemetry &&
+      this.batchedTelemetry.length > 0 &&
+      dateNow - this.batchedTelemetry[0].msInserted > this.sendPeriodMilliseconds
```

Comment:
Instead of `this.batchedTelemetry[0].msInserted`, shouldn't this use the last element? This'll emit an event every five seconds without waiting for a period of inactivity.

Reply:
"telemetry should be batched up and sent periodically after there has been no activity for a defined period of time - like maybe 5 seconds"

I see now that stated [goal](https://github.com/getappmap/appmap-js/pull/847#discussion_r1020316345) was to send data only after a period of inactivity.

The downside of sending data only after a period of inactivity is that if there's no inactivity it won't send data and possibly run out of memory (i.e. if the scanner is used in a high performance system that continually scans AppMaps that continually become available).  Whereas the current implementation guarantees data will always be sent periodically.

I'll change it to use the last element.

Reply:
Changed.